readme.md

```diff
@@ -8,54 +8,109 @@
 [![Backers][backers-badge]][collective]
 [![Chat][chat-badge]][chat]
 
-[**rehype**][rehype] plugin to add `id`s to headings.
+**[rehype][]** plugin to add `id`s to headings.
+
+## Contents
+
+*   [What is this?](#what-is-this)
+*   [When should I use this?](#when-should-i-use-this)
+*   [Install](#install)
+*   [Use](#use)
+*   [API](#api)
+    *   [`unified().use(rehypeSlug)`](#unifieduserehypeslug)
+*   [Types](#types)
+*   [Compatibility](#compatibility)
+*   [Security](#security)
+*   [Related](#related)
+*   [Contribute](#contribute)
+*   [License](#license)
+
+## What is this?
+
+This package is a [unified][] ([rehype][]) plugin to add `id`s to headings.
+It looks for headings (so `<h1>` through `<h6>`) that do not yet have `id`s
+and adds `id` attributes to them based on their textual content.
+The algorithm that generates `id`s is [`github-slugger`][github-slugger], which
+matches how GitHub works.
+
+**unified** is a project that transforms content with abstract syntax trees
+(ASTs).
+**rehype** adds support for HTML to unified.
+**hast** is the HTML AST that rehype uses.
+This is a rehype plugin that adds links to headings in the AST.
+
+## When should I use this?
+
+This plugin is useful when you have relatively long documents and you want to be
+able to link to particular sections.
+
+A different plugin, [`rehype-autolink-headings`][rehype-autolink-headings], adds
+links to these headings back to themselves, which is useful as it lets users
+more easily link to particular sections.
+
+This plugins matches how GitHub generates `id`s.
+If you want to generate them differently, you can fork this plugin: it’s pretty
+small!
```

Comment:
I wouldn’t suggest to create a fork. Forks lead to unmaintained low quality packages in the npm registry (which likely use this exact package’s metadata!).

I think it’s better to just omit this sentence.

If people ask for a feature to customize the generated id, this could be added as a feature, i.e. by accepting a function to generate it.

Reply:
Fork may be too much, but this plugin is like 5 lines. I specifically *don’t* see a need for customization (see existing issues: https://github.com/rehypejs/rehype-slug/issues?q=is%3Aissue).

How about:

```
If you want to generate them differently, you can create your own plugin based
on this one (which is only a few lines).
```

Reply:
Better, but still, personally I’d leave it out.